src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -490,4 +494,61 @@ default ItemAttributeModifiers getAttributeModifiers() {
 
         return CommonHooks.computeModifiedAttributes(self(), defaultModifiers);
     }
+
+    /**
+     * Wraps the original {@link ItemStack#save} encoding behavior and logs additional information if an exception is thrown.
+     */
+    default Tag encode(HolderLookup.Provider provider, Tag tag) {
+        try {
+            return self().CODEC.encode(self(), provider.createSerializationContext(NbtOps.INSTANCE), tag).getOrThrow();
+        } catch (Exception exception) {
+            logItemInfo(exception, tag);
+            throw exception;
+        }
+    }
+
+    /**
+     * Wraps the original {@link ItemStack#save} encoding behavior and logs additional information if an exception is thrown.
+     */
+    default Tag encode(HolderLookup.Provider p_332160_) {
+        try {
+            return self().CODEC.encodeStart(p_332160_.createSerializationContext(NbtOps.INSTANCE), self()).getOrThrow();
+        } catch (Exception exception) {
+            logItemInfo(exception, null);
+            throw exception;
+        }
+    }
+
+    /**
+     * Logs component information and tag data for an itemstack that failed to save.
+     * 
+     * <pre>
+     * Example:
+     * Error saving itemstack [1 minecraft:dirt]. Original cause: java.lang.NullPointerException
+     * With components:
+     * {
+     *    neoforge:test=>Test[s=null]
+     *    minecraft:max_stack_size=>64
+     *    minecraft:lore=>ItemLore[lines=[], styledLines=[]]
+     *    minecraft:enchantments=>ItemEnchantments{enchantments={}, showInTooltip=true}
+     *    minecraft:repair_cost=>0
+     *    minecraft:attribute_modifiers=>ItemAttributeModifiers[modifiers=[], showInTooltip=true]
+     *    minecraft:rarity=>COMMON
+     * }
+     * With tag: {}
+     * </pre>
+     */
+    private void logItemInfo(Exception original, @Nullable Tag tag) {
+        StringBuilder cause = new StringBuilder("Error saving itemstack [" + self() + "]. Original cause: " + original);
+
+        cause.append("\nWith components:\n{");
+        self().getComponents().forEach((component) -> {
+            cause.append("\n\t").append(component);
+        });
+        cause.append("\n}");
+        if (tag != null) {
+            cause.append("\nWith tag: ").append(tag);
+        }
+        LogManager.getLogger().error(cause.toString());
```

Comment:
Unsure if it be worth swapping this out for `Util.logAndPauseIfInIde`.
This would still log to console but, also if running in dev cause the ide to break at this point; This would however require devs to add breakpoints to `Util.doPause`.

```suggestion
        Util.logAndPauseIfInIde(cause.toString());
```